pkg/policy/api/utils.go

```diff
@@ -97,3 +101,155 @@ func ParseL4Proto(proto string) (L4Proto, error) {
 	p := L4Proto(strings.ToUpper(proto))
 	return p, p.Validate()
 }
+
+// GenerateToServiceRulesFromEndpoint populates egress rule with ToCIDR rules
+// based on ToServices defined in egress rule and provided endpoint
+func (e *EgressRule) GenerateToServiceRulesFromEndpoint(serviceInfo types.K8sServiceNamespace, endpoint types.K8sServiceEndpoint) error {
+	for _, service := range e.ToServices {
+		// TODO: match services by labels
+		if service.K8sService == serviceInfo {
+			if err := generateToCidrFromEndpoint(e, endpoint); err != nil {
+				return err
+			}
+			// TODO: generateToPortsFromEndpoint when ToPorts and ToCIDR are compatible
+		}
+	}
+	return nil
+}
+
+// generateToCidrFromEndpoint takes an egress rule and populates it with ToCIDR rules based on provided enpoint object
+func generateToCidrFromEndpoint(egress *EgressRule, endpoint types.K8sServiceEndpoint) error {
+	// This will generate one-address CIDRs consisting of endpoint backend ip
+	mask := net.CIDRMask(128, 128)
+	for ip := range endpoint.BEIPs {
+		epIP := net.ParseIP(ip)
+
+		found := false
+		for _, c := range egress.ToCIDR {
+			_, cidr, err := net.ParseCIDR(string(c))
+			if err != nil {
+				return err
+			}
+			if cidr.Contains(epIP) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			cidr := net.IPNet{IP: epIP.Mask(mask), Mask: mask}
+			egress.ToCIDR = append(egress.ToCIDR, CIDR(cidr.String()))
```

Comment:
Instead of doing this, you could re-write the function to return a slice of cidrs. This prevents to have something in the `egress.ToCIDR` in case something bad happens on line 131 for example.

If you return a slice of cidrs, and an error, you can check for the error and if the error is nil only then apply the slice to the `egress.ToCIDR`

Reply:
Makes sense, will di